linkcheck/robotparser2.py

```diff
@@ -158,13 +159,13 @@ def parse (self, lines):
                     entry = Entry()
                     state = 0
             # remove optional comment and strip line
-            i = line.find('#')
+            i = decode_for_unquote(line).find('#')
             if i >= 0:
                 line = line[:i]
             line = line.strip()
             if not line:
                 continue
-            line = line.split(':', 1)
+            line = decode_for_unquote(line).split(':', 1)
```

Comment:
This looks fine, although I question why the robots.txt files gets bytes as the input, and why is it hardcoding the charset to UTF-8 instead of relying on the Content-Type HTTP header.

Reply:
Is this as simple as passing `decode_unicode=True` to iter_lines() in RobotFileParser.read()?

Reply:
yes, great idea!